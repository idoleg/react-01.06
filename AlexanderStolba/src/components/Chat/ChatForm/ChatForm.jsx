import React, { useState } from 'react';
import './ChatForm.css';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

function useInput(initialState) {
    const [state, setState] = useState(initialState);

    const setInput = (event) => {
        setState(event.target.value);
    }

    return [state, setInput, setState];
}

export const ChatForm = ({ onSendMessage }) => {
    const [name, setName] = useInput("Alex");
    const [content, setContent, setContentState] = useInput("");

    const handleSubmit = (event) => {
        event.preventDefault();
        onSendMessage({ name, content });
        setContentState("");
    }
    
    const handleKeyUp = (event) => {

        if (event.keyCode === 13) {
            event.preventDefault();
            onSendMessage({ name, content });
            setContentState("");
        }
    }

    return (
        <form>
            <TextField 
            id="standard-basic"
            label="Name"
            placeholder="Name:"
            value={name}
            onChange={setName} 
            />
   
            <TextField
            id="standard-textarea"
            label="Content"
            multiline
            rows={4}
            placeholder="Your message:"
            value={content}
            onChange={setContent}
            className="second-textarea"
            onKeyUp={handleKeyUp}
            />
            
            <Button 
                variant="contained" 
                color="secondary" 
                onClick={handleSubmit}
            >
                Send
            </Button>
        </form>
    )
}

ChatForm.propTypes = {
    onSendMessage: PropTypes.func.isRequired,
}