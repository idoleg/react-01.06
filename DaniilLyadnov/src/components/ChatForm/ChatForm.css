.chat-form {
	width: 100%;
	background: #fff;
	padding: 12px 7px;
	box-sizing: border-box;
	flex-grow: 1;
}

.input-name,
.input-content {
	min-height: 35px;
}

.input-name {
	width: 150px;
	margin-right: 7px!important;
}

.input-content {
	width: calc(100% - 150px - 50px - 7px);
	margin-right: 12px!important;
}

.input-name label,
.input-content label {
	line-height: 0;
}

.input-name input,
.input-content .MuiOutlinedInput-multiline {
	padding: 8px;
}

.input-content textarea {
	max-height: 100px;
	overflow-y: auto!important;
}

.MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline {
	border-color: #2094e6!important;
	border-width: 1px!important;
}

.MuiFormLabel-root.Mui-focused {
	color: #2094e6!important;
}

.MuiIconButton-root {
	padding: 1px!important;
}

.MuiSvgIcon-root {
	font-size: 2rem!important;
	color: #2094e6!important;
}

.MuiIconButton-root:hover {
	background: unset!important;
}