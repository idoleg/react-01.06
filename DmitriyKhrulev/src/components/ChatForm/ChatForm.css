.send-form {
   display: flex;
   justify-content: space-between;
   flex-wrap: wrap;
   margin: 0 auto;
   width: 80%
}
.name-form {
   max-width: 150px;
   min-width: 70px;
}
.content-form {
   margin: 10px 0;

}