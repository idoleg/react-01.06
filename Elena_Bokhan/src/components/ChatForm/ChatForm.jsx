import React, {useState} from 'react';
import propTypes from 'prop-types';

function useInput(initialState){
	const [state, setState] = useState(initialState)

	const setInput =(event) =>{
		setState(event.target.value)
	}
	return [state,setInput,setState]
}
const ChatForm = (props) => {
	
	const [name, setName, setNameState] = useInput("Your name");
	const [text, setText, setTextState] = useInput("type your message");

	const handleSubmit = () =>{
		event.preventDefault();
		props.onSendMsg({name,text});
		props.addChat({name});
		setTextState("type your message");
		setNameState("Your name");
	}
	
	const handleKeyUp = (event) =>{
			if(event.keyCode === 13){
				props.onSendMsg({name,text});
				setTextState("type your message");
				setNameState("Your name");
		}
	}
	return (
		<form onSubmit = {handleSubmit} className ="chat-form">
			<input type="text"
				placeholder = "Your name"
				value = {name}
				onChange = {setName}
				onKeyUp = {handleKeyUp}/>
			<textarea 
				placeholder = "type your message"
				value = {text}
				onChange = {setText}
				onKeyUp = {handleKeyUp} >
			</textarea>
			<button >SEND</button>
		</form>
	)
}
export default ChatForm;