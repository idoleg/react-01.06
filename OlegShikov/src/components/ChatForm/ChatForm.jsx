import React from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import useInput from '../../hooks/useInput';

export const ChatForm = ({ onSendMessage }) => {
    const [name, setName] = useInput("Oleg");
    const [content, setContent, setContentState] = useInput("");

    const handleSubmit = (event) => {
        event.preventDefault();
        onSendMessage({ name, content });
        setContentState("");
    }

    return (
        <form>
            <TextField
                label="Name"
                variant="outlined"
                placeholder="Name"
                value={name}
                onChange={setName}
            />
            <TextField
                label="Content"
                variant="outlined"
                multiline
                rows={4}
                placeholder="Type text of message"
                value={content}
                onChange={setContent}
            />
            <Button 
                variant="contained" 
                color="primary" 
                onClick={handleSubmit}
            >
                Send
            </Button>
        </form>
    )
}

ChatForm.propTypes = {
    onSendMessage: PropTypes.func.isRequired,
}

// onChange={event => setContent(event.target.value)}
