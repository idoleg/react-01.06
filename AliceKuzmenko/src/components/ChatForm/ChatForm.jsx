import React, { useState } from 'react';
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import './ChatForm.css'

function useInput(initialState) {
    const [state, setState] = useState(initialState);

    const setInput = (event) => {
        setState(event.target.value);
    }

    return [state, setInput, setState];
}

export const ChatForm = ({ onSendMessage }) => {
    const [name, setName] = useInput("Me");
    const [content, setContent, setContentState] = useInput("");

    const handleChange = (event) => {
        this.setState({ value: event.target.value });
    }

    const handleSubmit = (event) => {
        //alert('Отправленное имя: ' + this.state.value);
        onSendMessage({ name, content});
        event.preventDefault();
        setContentState("");

    }
    const handleKeyUp = (event) => {
        //console.log("event")
        if (event.keyCode === 13) { // Enter
            handleSubmit(event);
        }
     };
    return (
        <form>
            <TextField
                label="Name"
                variant="outlined"
                placeholder="Name"
                value={name}
                onChange={setName}
            />
            <TextField
                label="Content"
                variant="outlined"
                multiline
                rows={4}
                placeholder="Your message"
                value={content}
                onChange={setContent}
                onKeyUp = { (event) => handleKeyUp(event) }
            />
            <Button 
                variant="contained" 
                color="primary" 
                onClick={handleSubmit}
                
            >
                Send
            </Button>
        </form>
    )
}

ChatForm.propTypes = {
    onSendMessage: PropTypes.func.isRequired,
}