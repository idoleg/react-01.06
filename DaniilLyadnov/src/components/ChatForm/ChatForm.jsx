import './ChatForm.css';
import React, {useState} from "react";
import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import IconButton from '@material-ui/core/IconButton';
import TelegramIcon from '@material-ui/icons/Telegram';
import SendRoundedIcon from '@material-ui/icons/SendRounded';

function useInput(initialState) {
    const [state, setState] = useState(initialState);

    const setInput = (event) => {
        setState(event.target.value);
    };

    return [state, setInput, setState];
}

export const ChatForm = ({onSendMessage}) => {
    const [name, setName] = useInput('');
    const [content, setContent, setContentState] = useInput('');

    const handleSubmit = (event) => {
        event.preventDefault();

        if (content.trim() !== '') {
            onSendMessage({name, content});
            setContentState('');
        }
    };

    const handleKeyUp = (event) => {
        if (event.keyCode === 13) {
            if (content.trim() !== '') {
                onSendMessage({name, content});
                setContentState('');
            }
        }
    };

    return (
        <form className={'chat-form'}>
            <TextField
                className={"input-name"}
                label="Name"
                variant="outlined"
                placeholder="Name"
                value={name}
                onChange={setName}
            />

            <TextField
                className={"input-content"}
                label="Content"
                multiline
                variant="outlined"
                placeholder="Type text of message"
                value={content}
                onChange={setContent}
                onKeyUp={handleKeyUp}
            />

            <IconButton
                aria-label="delete"
                onClick={handleSubmit}
            >
                <SendRoundedIcon />
            </IconButton>
        </form>
    )
};

ChatForm.propTypes = {
    onSendMessage: PropTypes.func.isRequired,
};