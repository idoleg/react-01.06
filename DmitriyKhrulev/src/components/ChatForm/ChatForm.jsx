import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button'
import TextField from '@material-ui/core/TextField'
import useInput from '../../hooks/useInput'
import './ChatForm.css'




export const ChatForm = ({ onSendMessage }) => {
   const [name, setName, setNameState] = useInput('Dima');
   const [content, setContent, setContentState] = useInput('')

   const handleSubmit = (e) => {
      e.preventDefault();
      onSendMessage({ name, content });
      setContentState('');
      // setNameState('');
   }

   const handleKeyUp = (e) => {
      if (e.keyCode === 13) {
         e.preventDefault();
         onSendMessage({ name, content });
         setContentState('');
         // setNameState('');
      }


   }

   return (
      <form className='send-form' onSubmit={handleSubmit}>
         <TextField className='name-form'
            id="outlined-basic"
            label="Name"
            variant="outlined"
            placeholder="Имя"
            value={name}
            onChange={setName}
         />

         <TextField
            onKeyUp={e => handleKeyUp(e, content)}
            className='content-form'
            id="outlined-multiline-static"
            label="Content"
            placeholder="Введите собщение"
            multiline
            rowsMax={4}
            value={content}
            onChange={setContent}
            variant="outlined"
         />
         <Button onClick={handleSubmit} variant='contained' color="primary"> Отправить </Button>
      </form>
   );
}

// ChatForm.propTypes = {
//    onSendMessage: PropTypes.func.isRequired,
// }