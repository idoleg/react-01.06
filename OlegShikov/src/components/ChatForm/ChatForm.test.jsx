import React from 'react';
import { ChatForm } from './ChatForm';
import { shallow } from 'enzyme';
import Button from '@material-ui/core/Button';

describe('ChatForm', () => {
    it('render content', () => {
        const element = shallow(<ChatForm onSendMessage={() => { }} />);
        expect(element).toMatchSnapshot();
    })

    it('simuldate events', () => {
        const name = "Ivan";
        const content = "Test";
        const props = {
            onSendMessage: jest.fn(),
        }

        const element = shallow(<ChatForm {...props} />);
        element.find({ label: "Name" }).at(0).simulate('change', { target: { value: name } })
        element.find({ label: "Content" }).at(0).simulate('change', { target: { value: content } })
        element.find(Button).simulate('click', { preventDefault: () => { } });

        expect(props.onSendMessage).toHaveBeenCalledTimes(1)
        expect(props.onSendMessage).toHaveBeenCalledWith({ name, content })
    })
}) 