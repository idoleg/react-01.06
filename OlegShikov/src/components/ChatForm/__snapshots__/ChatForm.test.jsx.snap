// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatForm render content 1`] = `
<form>
  <WithStyles(ForwardRef(TextField))
    label="Name"
    onChange={[Function]}
    placeholder="Name"
    value="Oleg"
    variant="outlined"
  />
  <WithStyles(ForwardRef(TextField))
    label="Content"
    multiline={true}
    onChange={[Function]}
    placeholder="Type text of message"
    rows={4}
    value=""
    variant="outlined"
  />
  <WithStyles(ForwardRef(Button))
    color="primary"
    onClick={[Function]}
    variant="contained"
  >
    Send
  </WithStyles(ForwardRef(Button))>
</form>
`;
